pkg/analysis/passes/gomanifest/gomanifest.go

```diff
@@ -141,18 +146,22 @@ func verifyManifest(manifest map[string]string, goFiles []string, sourceCodeDir
 			return err
 		}
 		// calculate the sha256sum of the go file
-		sha256sum, err := hashFileContent(goFilePath)
+		sha256sum, windowsSha256Sum, err := hashFileContent(goFilePath)
 		if err != nil {
 			return err
 		}
 		// check if the sha256sum is in the manifest
 		manifestSha256sum, ok := manifest[goFileRelativePath]
 
 		if !ok {
-			return fmt.Errorf("could not find file %s with hash %s in manifest", goFileRelativePath, sha256sum)
+			return fmt.Errorf(
+				"could not find file %s with hash %s in manifest",
+				goFileRelativePath,
+				sha256sum,
+			)
 		}
 		// check if the sha256sum in the manifest matches the calculated sha256sum
-		if sha256sum != manifestSha256sum {
+		if sha256sum != manifestSha256sum && windowsSha256Sum != manifestSha256sum {
```

Comment:
this is the fundamental change in the validation of the go manifest. now we calculate two hashes, one with linux line endings and other with windows line endings. We accept either as a valid hash.

Reply:
Should it be or instead of and?

Reply:
This is the condition to fail so passing one is a success (hence and `and`). If we would put an `or` here it'll always fail:

A file is hashed and written into the manifest. the hash is calculated based on the content that might be windows line end or linux line end.

We calculate both hashes of the original file. one with linux line ending and other with windows line ending.

if neither of those hashes matches the hash in the manifest we fail the check.